clang/lib/Frontend/HeaderIncludeGen.cpp

```diff
@@ -427,7 +458,19 @@ void HeaderIncludesDirectPerFileCallback::InclusionDirective(
   if (!FromFile)
     return;
 
-  Dependencies[*FromFile].push_back(*File);
+  FileEntryRef headerOrModule = *File;
+  if (ModuleImported && SuggestedModule) {
+    OptionalFileEntryRef ModuleMapFile = HSI.getModuleMap().getModuleMapFileForUniquing(SuggestedModule);
+    if (ModuleMapFile) {
+      headerOrModule = *ModuleMapFile;
+    }
+  }
+
+  Dependencies[*FromFile].push_back({
+    .location = Loc,
```

Comment:
I think these are called designated initializers and are only available in C++20. LLVM only supports C++17 right now, so I think you might need to use a different initialization syntax. Ditto below.